internal/service/datazone/asset_type.go

```diff
@@ -149,18 +150,12 @@ func (r *assetTypeResource) Create(ctx context.Context, req resource.CreateReque
 
 	out, err := conn.CreateAssetType(ctx, &in)
 	if err != nil {
-		resp.Diagnostics.AddError(
-			create.ProblemStandardMessage(names.DataZone, create.ErrActionCreating, ResNameAssetType, plan.Name.String(), err),
-			err.Error(),
-		)
+		smerr.AddError(ctx, &resp.Diagnostics, err, smerr.ID, plan.Name.String())
```

Comment:
Should the key of the key/value pair be `name` here? This might apply to all cases when the identifier has not been returned from the API yet.

```suggestion
		smerr.AddError(ctx, &resp.Diagnostics, err, "name" plan.Name.String())
```